Userland/Libraries/LibPDF/Renderer.cpp

```diff
@@ -892,7 +892,21 @@ PDFErrorOr<void> Renderer::show_image(NonnullRefPtr<StreamObject> image)
     auto image_bitmap = TRY(load_image(image));
     if (image_dict->contains(CommonNames::SMask)) {
         auto smask_bitmap = TRY(load_image(TRY(image_dict->get_stream(m_document, CommonNames::SMask))));
-        VERIFY(smask_bitmap->rect() == image_bitmap->rect());
+
+        // Make softmask same size as image.
+        // FIXME: The smask code here is fairly ad-hoc and incomplete.
+        if (smask_bitmap->rect() != image_bitmap->rect()) {
+            auto scaled_smask_bitmap = TRY(Gfx::Bitmap::create(smask_bitmap->format(), image_bitmap->size()));
+
+            // Scale to fit, preserving aspect ratio.
+            Gfx::IntRect destination_rect { {}, scaled_smask_bitmap->size() };
+            Gfx::IntRect source_rect { {}, smask_bitmap->size() };
+            Gfx::Painter painter(scaled_smask_bitmap);
+            painter.draw_scaled_bitmap(destination_rect, smask_bitmap, source_rect);
+
+            smask_bitmap = scaled_smask_bitmap;
```

Comment:
Seems like a bit of a manual implementation of `Bitmap::scaled`, is this just the same as?
```c++
auto scale_x = static_cast<float>(image_bitmap->width()) / smask_bitmap->width();
auto scale_y = static_cast<float>(image_bitmap->height()) / smask_bitmap->height();
smask_bitmap = TRY(smask_bitmap->scaled(scale_x, scale_y));
```

Reply:
I guess it is!

wdyt about making a version of `scaled()` or `scaled_to_size()` or what that takes a size instead of a factor?

Reply:
ha, I had a similar thought, "it'd be nice not to have to compute the scale manually". `scaled_to_size` sounds good to me.

Reply:
Why do we have both ` Bitmap::scaled(float, float)` and Painter's `do_draw_scaled_bitmap()` with independent implementations :/

Reply:
:man_shrugging: